### PR TITLE
feat(file-uploader): sync with @carbon/react v11

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

#10084 
### Description

Update file uploader 
### Changelog

**New**

- file-uploader-button
- add missing stories that are in react

**Changed**

- renamed file-uploader-drop-container to closely align to react
- update props and names to closely align to react

### CODE REVIEW

File: packages/carbon-web-components/src/components/file-uploader/defs.ts [hunk 1]
```diff
@@ -19,36 +19,30 @@ export enum FILE_UPLOADER_ITEM_STATE {
...
 export enum FILE_UPLOADER_ITEM_SIZE {
   /**
-   * Regular size.
+   * Small size.
    */
-  REGULAR = '',
+  SMALL = 'sm',
 
   /**
    * Small size.
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/file-uploader/defs.ts:40]:
```suggestion
   * Medium size.
```

File: packages/carbon-web-components/src/components/file-uploader/file-uploader-button.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        composed: true,
+        detail: {
+          addedFiles,
+        },
+      })
+    );
+
+    const fileInput = this?.shadowRoot?.querySelector(selectorInput);
+    if (fileInput) {
+      (fileInput as HTMLInputElement).value = ''; // carbon-web-components#904
```
Comment (kennylam) [inline packages/carbon-web-components/src/components/file-uploader/file-uploader-button.ts:71]: What's this comment mean?

Comment (ariellalgilmore) [reply packages/carbon-web-components/src/components/file-uploader/file-uploader-button.ts:71]: oh forgot to come back to that it was part of the original code..... but it has been [fixed](https://github.com/carbon-design-system/carbon-web-components/issues/904), so i'll remove this comment

Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/file-uploader/file-uploader-button.ts:71]:
```suggestion
      (fileInput as HTMLInputElement).value = '';
```

**Review by RichKummer — APPROVED** [review]
@ariellalgilmore this looks great! It looks like the react version has one extra style for `buttonKind` that doesn't appear in this build, which is `danger--primary`. Once you add that in, this will be perfect

Screen Shot 2023-04-25 at 4 14 30 PM

**Review by RichKummer — COMMENTED** [review]
@ariellalgilmore this looks great! It looks like the react version has one extra style for `buttonKind` that doesn't appear in this build, which is `danger--primary`. Once you add that in, this will be perfect

<img width="1316" alt="Screen Shot 2023-04-25 at 4 14 30 PM" src="https://user-

**Comment by ariellalgilmore** [comment]
Thanks @RichKummer ! just updated!

**Review by kennylam — APPROVED** [review]

**Review by RichKummer — APPROVED** [review]
🥳

**Pull request merged**